### DISCUSSION TITLE
Fundamental problems with 1.6.2

### DISCUSSION THREAD

**Discussion by hmmueller** [Paciente8159/uCNC#709] [Q&A] [answered]
My setup: Homebuilt standard CNC router, Arduino UNO, CNC Shield 3.0, all 3 motors are nicely and repeatably running with a test script that drives them directly. So hardware is fine.

Then, sorry, I'd like to run with Arduino IDE. This is the simplest thing - one tool that also (very important for me) has an integrated Serial Monitor - so when uploading a script worked, I know that the IDE and Arduino speak to each other.

This is the reason why I run with versions of µCNC of 1.7.2 and below - they compile small enough without any changes.

Yet:
1.7.2, when I upload it, does nothing at all. No message, no reaction to any command. So I gave it up.

1.6.2, on the other hand, does (or rather, did some) work. I could run it, it answered to ? and $$ and even G00 (changed internal state, but no motor moved, even though I know they work - see above).
However, after trying to change some flags I have now locked myself out: When I start or reset, I get (now ... not earlier):

18:36:00.826 -> Grbl 1.6 [uCNC v1.6.2 '$' for help]
18:36:00.913 -> [MSG:'$H'|'$X' to unlock]
18:36:00.913 -> ALARM:1
18:36:00.913 -> [MSG:Check Limits]
18:36:00.913 -> [MSG:Reset to continue]

I do have limit switches; but now I have them cut out and clamped the X, Y, and Z limit switch pins of the CNC Shield to ground with jumpers. Still, I get the message above.
(From some measurements with my RC electronics for my own limit switches I *guess* - but am not at all sure - that the Z limit switch pin is either set for OUTPUT instead of INPUT, or is in some weird state; instead of some 0.13V at the X and Y limit switch pins - behind my RC -, it showed a solid 0.00 V; not what I expected. Something is weird here, but I cannot look into the Arduino).

So, please, does anyone have an idea how I could continue?

(As a side question: Is there some sort of hook where one can write a piece of debug code that sends something back over the serial wire? E.g. like this - my problem is writing to the serial thing ...

```
int lastDebugOut = 0;
...
void loop() {
  cnc_run();
  if (millis() > lastDebugOut + 1000) {
    debugOut();
    lastDebugOut = millis();
  }

void debugOut() {
    ...
   Serial...write something I'd like to know...
}
```

Thanks a lot!!!
Harald M.

**Comment by Paciente8159** [comment, marked as answer]
Hi there.
My guess with v1.7.2 not responding (or apparently being unresponsive) is that the alarm state is active because of the ESTOP pin state.
You can confirm this by sending a `?` char.

The problem with 1.6.2 is probably because you are using the board configuration that is similar to grbl1.1. That version is not compatible with shield v3 because z limit pin and one of the pwm spindle pin was switched.
You need to use a mapping for shield v3.

You can use the current version of uCNC. It should still fit on the uno with a normal configuration.

Let me know if you run into problems.

**Comment by hmmueller** [comment]
Ok - I changed over to the master branch, and VSCode+PlatformIO. Works better, but not good enough ... separate discussion. I'll close this one.

Thanks!
H.M.
